### ISSUE TITLE
Improper date format (wrong locale?)

### ISSUE THREAD

**Issue by MagpieFourtyTwo** [cgeo/cgeo#8825]
**Describe the bug:**
If smartphone's ``Language`` is set to ``English``, c:geo uses the date format ``DD\MM\YYYY``, although ``Region`` is set to ``Germany``. But as far as I understand ``localtime()``, date and number representation should follow ``Region``, thus should be ``DD.MM.YYYY`` (following ISO 8601). This should be changed, cause the currently used backslashes give reason to believe it's ``MM\DD\YYYY`` (just fell for this recently ;) ).

**To Reproduce:**
1. set your phone's ``Language`` to ``English``
2. set your phone's ``Region`` to ``German``
3. have a look at e. g. c:geo's Logbook page or do ``Log Visit`` and select a date more than two days away

**Actual behavior/state after performing these steps:**
Dates are written as ``DD\MM\YYYY`` instead of expected ``DD.MM.YYYY``

**Expected behavior/state after performing these steps:**
Date representation should follow ISO 8601.

**Is the problem reproducible:**
Yes

**Additional context:**
Long date format should be ``D. MMMM YYYY`` - but c:geo does not put the dot (in e. g. signature template [DATE]).  Maybe this is an Android bug (as in Android's system settings the date example is missing the dot, too), but perhaps it can be added by c:geo ...?

**Comment by Lineflyer** [comment]
Which "English" did you set on your smartphone? AFAIK there should be country distinctions available such as "US", "UK", etc.

**Comment by MagpieFourtyTwo** [comment]
Yep, that was my second hope - but over here there are just two English: plain ``English`` and ``English (Zawgyi)``

**Comment by eddiemuc** [comment]
@MagpieFourtyTwo is this issue still relevant?

**Comment by MagpieFourtyTwo** [comment]
Yes, it is, most regretfully, even in both respects, i. e. concerning the logbook's date format as well as the macro's substitution.

**Comment by bekuno** [comment]
@MagpieFourtyTwo Please check if this issue is further relevant.

**Comment by MagpieFourtyTwo** [comment]
Part of it - short date format (DD.MM.YYYY) is correct now, only the dot in "DD. MMMM YYYY" is still missing.

**Comment by ztNFny** [comment]
When writing the full month name it's most common not to put a dot behind the day # in most countries - Germany is among the few putting it commonly.
https://en.m.wikipedia.org/wiki/List_of_date_formats_by_country

**Comment by ztNFny** [comment]
Sent to early ...
If Android is returning a wrong date format it should be addressed in Android, not in an app. 

Therefore we shouldn't change it in cgeo as we'd be breaking it for most countries. Also that'd make that 1 app appear different than all other apps, making it appear "broken" to the user, even if it might be the only right one.
